src/Datadog.Trace.ClrProfiler.Managed/Emit/MethodBuilder.cs

```diff
@@ -350,51 +350,43 @@ private MethodInfo TryFindMethod()
             // prevent multiple enumerations
             var methodEnumerable = methods.AsEnumerable();
 
-            if (_methodBase != null)
-            {
-                // For the case I haven't encountered, where MethodBase is not MethodInfo
-                methodEnumerable =
-                    methodEnumerable.Where(
-                        mi => mi.IsGenericMethod == _methodBase.IsGenericMethod
-                           && mi.IsAbstract == _methodBase.IsAbstract
-                           && mi.IsPrivate == _methodBase.IsPrivate
-                           && mi.IsPublic == _methodBase.IsPublic
-                           && mi.IsVirtual == _methodBase.IsVirtual
-                           && mi.Name == _methodBase.Name);
-            }
-            else
-            {
-                // A legacy fallback attempt to match on the concrete type
-                methodEnumerable =
-                    methodEnumerable
-                       .Where(mi => mi.Name == _methodName)
-                       .Where(mi => _returnType == null || mi.ReturnType == _returnType);
-            }
+            // A legacy fallback attempt to match on the concrete type
+            methodEnumerable =
+                methodEnumerable
+                   .Where(mi => mi.Name == _methodName)
+                   .Where(mi => _returnType == null || mi.ReturnType == _returnType);
```

Comment:
[low-priority performance nit] this iterates the enumerable twice. next time we modify this code, we should merge the two `Where()` calls into one:
```csharp
.Where(mi => mi.Name == _methodName && (mi => _returnType == null || mi.ReturnType == _returnType))
```

Reply:
Sorry, couldn't help myself :)
Actually, just tacking on a LINQ `Where(...)` call does not enumerate the `IEnumerable`. The real enumeration happens later, when you call `ToArray()`.

The `Where<T>(this IEnumerable<T> source, Func<T, bool> predicate)` extension method returns a new instance of the `WhereEnumerableIterator<T>(source, predicate)` which only checks an element with the predicate when triggered by an enumeration of the collection, typically triggered by a call to `ToArray()`, `ToList()` or `foreach`.

Reply:
Yes, you're right. I used the wrong term. I didn't mean it "iterates" twice. What I meant is that in LINQ to Objects, it will tack on two separate `Func<MethodInfo, bool>` delegates to the enumerable. Invoking an additional delegate has quite a bit of overhead  impaired to the (possibly short-circuited) Boolean comparisons.